Civi/Sqltasks/Actions/RunSQLTask.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Civi\Sqltasks\Actions;
+
+use \Civi\ActionProvider\Action\AbstractAction;
+use \Civi\ActionProvider\Exception\ExecutionException;
+use \Civi\ActionProvider\Parameter\ParameterBagInterface;
+use \Civi\ActionProvider\Parameter\SpecificationBag;
+use \Civi\ActionProvider\Parameter\Specification;
+
+use CRM_Sqltasks_ExtensionUtil as E;
+
+class RunSQLTask extends AbstractAction {
+
+  /**
+   * Run the action
+   *
+   * @param ParameterInterface $parameters
+   * @param ParameterBagInterface $output
+   * @return void
+   */
+  protected function doAction(ParameterBagInterface $parameters, ParameterBagInterface $output) {
+    // Configuration
+    $task_id = $this->configuration->getParameter('task_id');
+    $log_to_file = $this->configuration->getParameter('log_to_file');
+    $return_parameter = $this->configuration->getParameter('return_parameter');
+
+    // Parameters
+    $input_values = $parameters->getParameter('input_values');
+
+    try {
+      civicrm_api3('Sqltask', 'get', [ 'id' => $task_id ]);
+    } catch (\Exception $ex) {
+      throw new ExecutionException(E::ts("Task with ID '$task_id' not found"));
+    }
+
+    $exec_params = [
+      'id'          => $task_id,
+      'input_val'   => $input_values,
```

Comment:
I had to add `json_encode` here to get it to work with more than one input (otherwise it would just attempt to cast the array to string in some places, and fail in others where arrays are not accepted).

I suppose we could also look into fixing this in `CRM_Sqltasks_Task::execute()` instead, similar to your fix in `CRM_Sqltasks_BAO_SqltasksExecution::create()`. Would need a bit more testing to ensure we remain backwards-compatible.

(Handling of JSON parameters vs array parameters in API3 is very confusing to me. 🤯)